schema/pgsql/schema.sql

```diff
@@ -278,3 +278,32 @@ CREATE TABLE incident_history (
     CONSTRAINT pk_incident_history PRIMARY KEY (id),
     FOREIGN KEY (incident_id, rule_escalation_id) REFERENCES incident_rule_escalation_state(incident_id, rule_escalation_id)
 );
+
+CREATE OR REPLACE FUNCTION now_extended() RETURNS bigint
+    LANGUAGE SQL AS
+$$
+SELECT ((EXTRACT(EPOCH FROM CURRENT_TIMESTAMP) * 1_000) :: bigint);
+$$;
```

Comment:
Yes as the default should store the current unix timestamp with milliseconds precision to comply with the data timestamps used at other places in the icinga2 databases.
PostgreSQL does not allow to use subqueries in the default expression [(see this)](https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-PARMS-DEFAULT), which is why the workaround by using a function seems mandatory.

Reply:
I don't see the need for a subquery here. My PostgreSQL server doesn't seem to like `1_000` though.
```
jbrost=# create table foo (x bigint default EXTRACT(EPOCH FROM CURRENT_TIMESTAMP) * 1000);
CREATE TABLE
jbrost=# insert into foo (x) values (default);
INSERT 0 1
jbrost=# insert into foo (x) values (default);
INSERT 0 1
jbrost=# select * from foo;
       x       
---------------
 1701179227054
 1701179227894
(2 rows)
```

Reply:
Good catch! Didn't know that it would support extract commands in the default expression.
And yes, the decimal separator is usually only allowed in query contexts (which in this case will not work as we are in a non-query context).

Will change the schema to get rid of the function definition.